### PR TITLE
VDS-3139 Implement the export of products properties

### PR DESCRIPTION
**Pull request by trueboroda**
Description: 

--

QA-test: VDS-3142

Demo-test: VDS-3143

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.CatalogExportImport_1.1.0-pr-27-d9eddc3a.zip

### CODE REVIEW

File: src/VirtoCommerce.CatalogExportImportModule.Data/ExportImport/GenericTypeWithPropertiesClassMap.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        {
+            AutoMap(new CsvConfiguration(CultureInfo.InvariantCulture) { Delimiter = ";" });
+
+            var typeHasProperties = ClassType.GetInterfaces().Contains(typeof(IHasProperties));
+
+            if (!properties.IsNullOrEmpty() && typeHasProperties)
+            {
+                AddPropertyColumnDefinitionAndWritingMap(properties);
+
+                AddPropertyReadingMap(properties, propertyDictionaryItems);
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Data/ExportImport/GenericTypeWithPropertiesClassMap.cs]: May be just export/import writing/reading map?

File: src/VirtoCommerce.CatalogExportImportModule.Data/ExportImport/GenericTypeWithPropertiesClassMap.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                                .Distinct()
+                                .ToArray();
+                        }
+                        else
+                        {
+                            valuePropertyValues = valueProperty.Values?
+                                .Where(x => x.Value != null)
+                                .Select(x => x.Value.ToString())
+                                .ToArray();
+                        }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Data/ExportImport/GenericTypeWithPropertiesClassMap.cs]: The only difference is Distinct, may be merge them?

File: src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs [hunk 3]
```diff
@@ -1,16 +1,94 @@
...
+            return classMap;
+        }
+
+        private async Task<Property[]> LoadPropertiesAsync(string catalogId, string[] itemIds, string[] categoryIds)
         {
+            var result = Array.Empty<Property>();
+
+            var comparer = AnonymousComparer.Create((Property x) => x.Name);
+
+            if (categoryIds.IsNullOrEmpty() && itemIds.IsNullOrEmpty())
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs:50]: Empty Category ID & Outer ID is invalid state and will be put to the report

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs (hunk 3 above)
Comment (trueboroda) [reply src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs:50]: It is the request data.

**Review by asvishnyakov — APPROVED** [review]

File: src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.CatalogExportImportModule.Data/Services/PhysicalProductDataExporter.cs:50]: Yeah. it's export, sorry :)

**Pull request merged**